### PR TITLE
Mitigate the Lens disappearing with "Show Unit Path on Hover" setting

### PR DESCRIPTION
**Pull request by the-m4a**
This isn't a perfect solution, but I think it is reasonable and mitigates the issue of losing the lens entirely (you can see the gif below drawing the path up and to the left for the knight unit).
Lens mitigation with path on hover

As mentioned / shown in the issue thread #94 -- the "unit path" is its own lens, and there appears to be a limited toolset to work with:
- Using Layers with the "Movement" lens and whatever other lens happened to be showing did not function like I think we would hope it would
- There does not appear to be a way to get the actual hash of the currently applied lens, only a push/pop mechanism of saving and restoring whatever lens happens to be applied (_UILens.SaveActiveLens()_ does not return a value)
- There is "something" there with the Religion lens that makes it appear to sort of work, but that may be just more hacky than any of us would want to implement.  It appears to just leave the hexes colored but does not actually have the lens in place anymore; so it may be the religion lens that broken, not me.

I tested with vanilla, XP1, and XP2.  If when hovering on a unit with a path (destination), whether that be valid, invalid, or in unexplored territory, it will do like what is illustrated in the gif above.  I did verify that no matter the lens picked, the interfacemode was _VIEW_MODAL_LENS_

### CODE REVIEW

**Comment by Infixo** [comment]
Lenses are super wonky and difficult to work with, I agree.

>     * There does not appear to be a way to get the actual hash of the currently applied lens, only a push/pop mechanism of saving and restoring whatever lens happens to be applied (_UILens.SaveActiveLens()_ does not return a value)
Yup, there is not. Had the same issue in the past. This simple function (i.e. what lens is currently on) would make life much easier.


>     * There is "something" there with the Religion lens that makes it appear to sort of work, but that may be just more hacky than any of us would want to implement.  It appears to just leave the hexes colored but does not actually have the lens in place anymore; so it may be the religion lens that broken, not me.
There are some lenses (Religion is a good example) that are implemented by the game engine. I suppose they use then a separate "layer" and probably that is why you can still utilize "custom" layer and lenses.

**Comment by the-m4a** [comment]
@Infixo is there something you're wanting me to do here, or?  It's not clear.

**Comment by Infixo** [comment]
Nothing particular. I don’t know what to do with the PR. It doesn’t solve the issue, only makes things even more obscure. I think I would leave the issue as it is until we find a proper way to solve it. What do the others think?

**Comment by the-m4a** [comment]
We could just disable the show unit path on hover if a lens is currently showing... that or what is in this PR are the only reasonable ways I have seen as of yet.

**Comment by Infixo** [comment]
I’d rather go for disabling one while other is active.
But to be clear what I mean.
1. In the world view, when City Mgmt is not open, then both of them should work as both ace activated on hover. One over banner, the other over a unit.
2. When City Mgmt is open, then one is permanent and then Unit Path should be disabled. Which makes sense because when I open city view, it is to manage city not units. And this will eliminate the blinking which is pretty annoying.

**Comment by alimulhuda** [comment]
I think this fix doesn't have anything to do with whether or not city management is open. The hover over path display is already disabled when city management mode is turned on and also when the district/city ranged attack is being executed (Line 391 in the file).
I've seen the thing happen when say I'm using the city overlap lens and planning to place districts. Then I happen to mouse over a unit which has a queued move order. Then I just lose the lens I'm operating in.
Just listing the issue reproduction steps below to explain what I mean:
1. Tell a unit to move very far, so it takes multiple turns to move.
1. Deselect the unit
1. Mouse over the unit's icon (It should show the queued path only while the mouse is placed over the icon)
1. Turn on any lens
1. Mouse over the unit's icon again
1. Move the mouse so it's not on the unit's icon
At this point without the fix it does not go back to the lens used in step 4. With the fix the lens is temporarily disabled while the unit path is shown and then when the mouse moves away the display shows the lens used in step 4.

The push/pop fix seems to work nicely. I think it's a matter of personal preference between the choice of push/pop vs disable. @Infixo Did you notice any issues with the push/pop fix?

File: Assets/UI/worldinput_CQUI.lua [hunk 1]
```diff
@@ -431,6 +434,13 @@ function RealizeMovementPath(showQueuedPath:boolean, unitID:number)
 
     -- Only update if a new unit or new plot from the previous update.
     if m_cachedPathUnit ~= kUnit or m_cachedPathPlotId  ~= endPlotId then
+        -- BEGIN CQUI changes from basegame RealizeMovementPath (2nd of 2 in this function)
+        if (CQUI_im == InterfaceModeTypes.VIEW_MODAL_LENS) then
+            -- Save the currently displayed lens so that when it's either determined that the mouse is no longer hovering on a unit OR the unit movement path lens does not need to be shown, the current one can be retored
+            CQUI_SaveActiveLens();
```
Comment (alimulhuda) [inline Assets/UI/worldinput_CQUI.lua:440]: I would split the comment into two lines to make it easier to read

**Comment by Infixo** [comment]
I am objecting it. It is like I said - it doesn't fully the issue, kind of partial fix. So let it be.

**Pull request merged**